### PR TITLE
ci: Validate docs build in CI

### PR DESCRIPTION
**Pull request by vipulgupta2048**
:hatching_chick: Solves #169

Some info on options selected, 
`-n`  is to get all the warnings (Be nit-picky)
`-W` is to turn warning into errors

Also, added link-checking as well. Figure it would help in the long run. 

Signed-off-by: Vipul Gupta (@vipulgupta2048) <vipulgupta2048@gmail.com>

### CODE REVIEW

**Comment by vipulgupta2048** [comment]
Solves #169

**Comment by vipulgupta2048** [comment]
Warnings coming up now.
- `WARNING: missing attribute mentioned in :members: or __all__: module spidermon.contrib.scrapy.monitors, attribute `

- `/home/vipulgupta2048/spidermon/spidermon/contrib/scrapy/monitors.py:docstring of spidermon.contrib.scrapy.monitors.SpiderCloseMonitorSuite:13: WARNING: Duplicate ID: "module-spidermon.contrib.scrapy.monitors".`

- `/home/vipulgupta2048/spidermon/docs/source/monitors.rst:92: WARNING: py:class reference target not found: spidermon.core.monitors.Monitor
`

**Comment by Gallaecio** [comment]
The warnings should be fixed before these changes are merged. @vipulgupta2048 Do you think you can open a separate pull request to fix those warnings?

**Comment by vipulgupta2048** [comment]
Aye aye @Gallaecio, will get to it. 
Should we just ignore the warnings by placing them in `nitpick_ignore`, or resolve them because after my initial search I see that some are bugs.

**Comment by Gallaecio** [comment]
> Aye aye @Gallaecio, will get to it.
> Should we just ignore the warnings by placing them in `nitpick_ignore`, or resolve them because after my initial search I see that some are bugs.

Definitely fix the underlying causes, not ignore them :slightly_smiling_face:

File: .travis.yml [hunk 1]
```diff
@@ -14,6 +14,7 @@ install:
   - pip install tox-travis
 script:
   - tox
+  - tox -e docs
```
[the remark this answers is not in the document]
Comment (vipulgupta2048) [reply .travis.yml]: Oh! Good to know. Will get to it. Thanks @rennerocha

**Comment by vipulgupta2048** [comment]
Opening a new PR for fixing docs, and pushed a commit to build docs in python 3.7 only.

**Comment by vipulgupta2048** [comment]
Since #218 is now ready to go. I guess @Gallaecio we can review and approve this as well.

**Pull request merged**